### PR TITLE
#1327 work package redesign details

### LINKED ISSUE #1327: Work Package Redesign - Details Section
### Description

We will be working off of the current edit work package form to redesign the work package form

### Acceptance Criteria

Design looks like mock

### Proposed Solution

Edit the work package edit details file to appear like the mock. Change the name of the component to WorkPackageFormDetails

Use a grid for layout management to easily adjust fields for smaller screens

work off the feature/Workpackage-Form-Redesign Branch

### Mocks

Old Version 
Screen Shot 2023-09-05 at 6 54 41 PM

New Version
Screen Shot 2023-09-05 at 6 55 10 PM

epic #1318

### PR DESCRIPTION
**Pull request by RajinCoder**
## Changes

Changed WorkPackageEditDetails to WorkPackageFormDetails, changed all references to old name to match, modified fields inside of said file to match mock-up, and manipulated the NERAutocomplete and ChangeRequestDropdown file's height and width to match mock and also removed page block.
## Notes

I think a potential issue is that I might need to make the changes I did in autocomplete and the dropdown optional inorder to not mess up any other page's design.
## Screenshots

Screenshot 2023-11-01 at 10 00 23 PM
Screenshot 2023-11-01 at 10 00 10 PM

UPDATE:
Screenshot 2023-11-05 at 7 19 26 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)
Closes #1327

### CODE REVIEW

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx [hunk 1]
```diff
@@ -216,24 +216,24 @@ const WorkPackageEditContainer: React.FC<WorkPackageEditContainerProps> = ({ wor
             />
           </FormControl>
         </PageBlock>
-        <Typography variant="h5">Expected Activities</Typography>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx:219]: you shouldn't need to change any of this list stuff

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx [hunk 2]
```diff
@@ -171,7 +171,7 @@ const WorkPackageEditContainer: React.FC<WorkPackageEditContainerProps> = ({ wor
         { name: 'Projects', route: routes.PROJECTS },
         { name: `${projectWbsString} - ${workPackage.projectName}`, route: `${routes.PROJECTS}/${projectWbsString}` }
       ]}
-      headerRight={<ChangeRequestDropdown control={control} name="crId" />}
+      headerRight={<></>}
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx:174]: you can remove this line

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx [hunk 3]
```diff
@@ -126,8 +133,10 @@ const WorkPackageEditDetails: React.FC<Props> = ({
             value={userToOption(usersForProjectLead.find((user) => user.userId.toString() === lead))}
           />
         </Grid>
-        <Grid item xs={12} md={6} sx={{ mt: 1 }}>
+
+        <Grid item xs={12} md={4} sx={{ mt: 1 }}>
           <FormLabel>Project Manager</FormLabel>
+
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx:139]: delete this line

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx [hunk 4]
```diff
@@ -126,8 +133,10 @@ const WorkPackageEditDetails: React.FC<Props> = ({
             value={userToOption(usersForProjectLead.find((user) => user.userId.toString() === lead))}
           />
         </Grid>
-        <Grid item xs={12} md={6} sx={{ mt: 1 }}>
+
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx:136]: delete this line

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx [hunk 5]
```diff
@@ -64,9 +65,12 @@ const WorkPackageEditDetails: React.FC<Props> = ({
   );
 
   return (
-    <PageBlock title="Work Package Details">
+    <Box>
+      <Typography variant="h5" sx={{ marginBottom: '10px', color: 'white' }}>
+        Project Details
+      </Typography>
       <Grid container xs={12}>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx:72]: you can put spacing={1} here and remove all the sx={mt, mr} below

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx [hunk 6]
```diff
@@ -77,6 +81,12 @@ const WorkPackageEditDetails: React.FC<Props> = ({
             />
           </FormControl>
         </Grid>
+        <Grid item xs={12} md={2} lg={2} sx={{ mt: 2, mr: 2 }}>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx:84]: the way grid works is that there are 12 slots available in a row, so make sure to use them all up (especially in a form like this). I'd make this one md={3} and remove the lg option and do the same thing for the stage select grid item. That should allow the inputs to take up more of the screen

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx [hunk 7]
```diff
@@ -114,7 +121,7 @@ const WorkPackageEditDetails: React.FC<Props> = ({
             />
           </FormControl>
         </Grid>
-        <Grid item xs={12} md={6} sx={{ mt: 1 }}>
+        <Grid item xs={12} md={4} sx={{ mt: 1 }}>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx:124]: change the grid items for Project Lead and Project Manager to be md={5} to take up the whole screen

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx [hunk 8]
```diff
@@ -126,8 +133,10 @@ const WorkPackageEditDetails: React.FC<Props> = ({
             value={userToOption(usersForProjectLead.find((user) => user.userId.toString() === lead))}
           />
         </Grid>
-        <Grid item xs={12} md={6} sx={{ mt: 1 }}>
+
+        <Grid item xs={12} md={4} sx={{ mt: 1 }}>
           <FormLabel>Project Manager</FormLabel>
+
           <NERAutocomplete
             sx={{ mt: 1, width: '90%' }}
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx:141]: you should remove this line as well

**Review by RChandler234 — COMMENTED** [review]
basically, just a bunch of comments about spacing/ making the code cleaner:
it should end up looking like this:
image

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx (hunk 5 above)
Comment (RajinCoder) [reply src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx:72]: I tried putting spacing in but it seems to not be a property of Typography and could you clarify what you mean when you say to remove all the sx={mt, mr}

Comment (RChandler234) [reply src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx:72]: put it on the grid, not the container

Comment (RChandler234) [reply src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageFormDetails.tsx:72]: when I say remove that sx prop, I'm referring to where it's used on all the grid items below

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**